ocaml/driver/main_args.ml

```diff
@@ -716,11 +716,18 @@ let mk_dump_into_file f =
 ;;
 
 let mk_extension f =
-  "-extension", Arg.String f, "<extension> Enable the extension"
+  let available_extensions =
+    Clflags.Extension.(List.map to_string all)
+  in
+  "-extension", Arg.Symbol (available_extensions, f),
+  "<extension> Enable the extension (may be specified more than once)"
 ;;
 
-let mk_standard f =
-  "-standard", Arg.Unit f, " Disable all default extensions"
+let mk_no_extensions f =
+  "-no-extensions", Arg.Unit f,
+  " Disable all extensions, including extensions that are enabled by default,\n\
+  \    or specified in command line using -extension flags\n\
+  \    (before or after -no-extensions), or in OCAMLPARAM."
```

Comment:
I had to read this documentation a couple times to be sure that it was clarifying where extensions were ignored instead of offering exceptions.  Here's a possible rewrite, although it could maybe still use some tweaking:

```suggestion
let mk_no_extensions f =
  "-no-extensions", Arg.Unit f,
  " Disable all extensions, wherever they are specified; this flag\n\
  \    overrides the -extension flag (whether specified before or after this\n\
  \    flag), disables any extensions that are enabled by default, and\n\
  \    ignores any extensions requested in OCAMLPARAM."
```